Source/drlg_l2.cpp

```diff
@@ -2717,43 +2717,23 @@ static bool CreateDungeon()
 
 	for (int j = 0; j < DMAXY; j++) {     /// BUGFIX: change '<=' to '<' (fixed)
 		for (int i = 0; i < DMAXX; i++) { /// BUGFIX: change '<=' to '<' (fixed)
-			if (predungeon[i][j] == 67) {
-				predungeon[i][j] = 35;
-			}
-			if (predungeon[i][j] == 66) {
-				predungeon[i][j] = 35;
-			}
-			if (predungeon[i][j] == 69) {
-				predungeon[i][j] = 35;
-			}
-			if (predungeon[i][j] == 65) {
+			if (IsAnyOf(predungeon[i][j], 67, 66, 69, 65)) {
 				predungeon[i][j] = 35;
 			}
 			if (predungeon[i][j] == 44) {
 				predungeon[i][j] = 46;
-				if (predungeon[i - 1][j - 1] == 32) {
-					predungeon[i - 1][j - 1] = 35;
-				}
-				if (predungeon[i - 1][j] == 32) {
-					predungeon[i - 1][j] = 35;
-				}
-				if (predungeon[i - 1][1 + j] == 32) {
-					predungeon[i - 1][1 + j] = 35;
-				}
-				if (predungeon[i + 1][j - 1] == 32) {
-					predungeon[i + 1][j - 1] = 35;
-				}
-				if (predungeon[i + 1][j] == 32) {
-					predungeon[i + 1][j] = 35;
-				}
-				if (predungeon[i + 1][1 + j] == 32) {
-					predungeon[i + 1][1 + j] = 35;
-				}
-				if (predungeon[i][j - 1] == 32) {
-					predungeon[i][j - 1] = 35;
-				}
-				if (predungeon[i][j + 1] == 32) {
-					predungeon[i][j + 1] = 35;
+				for (int a = -1; a <= 1; a++) {
+					for (int b = -1; b <= 1; b++) {
+						if (a == 0 && b == 0)
+							continue;
+						if (i + a < 0 || j + b < 0)
+							continue;
+						if (i + a >= DMAXX || j + b >= DMAXY)
+							continue;

```

Comment:
Possibly loop over an array of Displacements instead.
```cpp
constexpr Displacement offsets[] = { { -1, -1 }, { -1, 0 }, ... };
for (Displacement offset : offsets) {
	if (predungeon[i + offset.x][j + offset.y] == 32) {
		predungeon[i + offset.x][j + offset.y] = 35;
	}
}
````

Reply:
I'd leave it for later when we have some mechanism for looping all surrounding tiles

Reply:
what you have here is pretty cryptic though with multiple loops and skip conditions.